lib_eio_linux/eio_linux.ml

```diff
@@ -1205,6 +1205,11 @@ class dir ~label (fd : dir_fd) = object
     | Dir_fd -> Some fd
     | _ -> None
 
+  method size path =
+    wrap_errors path @@ fun () ->
+    let st = Unix.stat path in
```

Comment:
This needs to be relative to `fd`.

Reply:
We could, but then it becomes a pain as you need to open just to get its size, but no strong feelings.

Reply:
Well, it needs to be correct and give you the size of the file you asked for, not some other file with the same name!

But `fstatat` and `statx` both take the FD as an argument anyway.

Reply:
ah yes I see what you mean, ofc, so I openat (resolve_beneath) -> fstat -> close ?

Reply:
Yes, I think that's best. It could be optimised for the common case where `fd` is `Fs`. Or, if we had linked uring requests, it could all be done in one go too.